multimodal/health_multimodal/image/model/model.py

```diff
@@ -0,0 +1,167 @@
+
+
+class ImageModel(nn.Module):
+    """
+    Image encoder module
+    """
+
+    def __init__(self,
+                 img_model_type: str,
+                 joint_feature_size: int,
+                 freeze_encoder: bool = False,
+                 **downstream_classifier_kwargs: Any):
+        super().__init__()
+
+        # Initiate encoder, projector, and classifier
+        self.encoder = ImageEncoder(img_model_type)
+        self.feature_size = get_encoder_output_dim(self.encoder)
+        self.projector = MLP(self.feature_size, output_dim=joint_feature_size,
+                             hidden_dim=joint_feature_size, use_1x1_convs=True)
+        self.downstream_classifier_kwargs = downstream_classifier_kwargs
+        self.classifier = self.create_downstream_classifier() if downstream_classifier_kwargs else None
+
+        # Initialise the mode of modules
+        self.freeze_encoder = freeze_encoder
+        self.train()
+
+    def train(self, mode: bool = True) -> Any:
+        super().train(mode=mode)
+        '''
+        Called to switch the model between training and evaluation modes.
+        '''
+        if self.freeze_encoder:
+            self.encoder.train(mode=False)
+            self.projector.train(mode=False)
+        return self
+
+    def forward(self, x: torch.Tensor, *args, **kwargs) -> ImageModelOutput:    # type: ignore
+        """
+        :param x: Input image tensor
+        """
+        with torch.set_grad_enabled(not self.freeze_encoder):
+            patch_x, pooled_x = self.encoder(x, return_patch_embeddings=True)
+            projected_patch_embeddings = self.projector(patch_x)
+            projected_global_embedding = torch.mean(projected_patch_embeddings, dim=(2, 3))
+
+        logits = self.classifier(pooled_x) if self.classifier else None
+        return ImageModelOutput(img_embedding=pooled_x,
+                                patch_embedding=patch_x,
+                                class_logits=logits,
+                                projected_patch_embeddings=projected_patch_embeddings,
+                                projected_global_embedding=projected_global_embedding)
+
+    def create_downstream_classifier(self, **kwargs: Any) -> ClassifierHeadType:
+        """
+        Creates the classification module for the downstream task
+        """
+        downstream_classifier_kwargs = kwargs if kwargs else self.downstream_classifier_kwargs
+        return MTModel(self.feature_size, **downstream_classifier_kwargs)
+
+    @torch.no_grad()
+    def get_patchwise_projected_embeddings(self, input_img: torch.Tensor, normalize: bool) -> torch.Tensor:
+        """
+        Get patchwise projected embeddings from the CNN model.
+        Args:
+            input_img: input tensor image [B, C, H, W]
+            normalize: If set to True, the embeddings are l2-normalized.
+        Returns:
+            projected_embeddings: tensor of embeddings in shape [batch, n_patches_h, n_patches_w, feature_size]
```

Comment:
This is Google style. We must use NumPy style for consistency.

Also, there must be a blank line between these and the first line.